src/services/navigationBar.ts

```diff
@@ -176,23 +190,35 @@ namespace ts.NavigationBar {
             }
         }
 
+        function hasNamedFunctionDeclarations(nodes: NodeArray<Statement>) {
+            if (forEach(nodes, s => s.kind === SyntaxKind.FunctionDeclaration && !isEmpty((<FunctionDeclaration>s).name.text))) {
+                return true;
+            }
+        }
+
         function isTopLevelFunctionDeclaration(functionDeclaration: FunctionLikeDeclaration) {
             if (functionDeclaration.kind === SyntaxKind.FunctionDeclaration) {
                 // A function declaration is 'top level' if it contains any function declarations 
                 // within it. 
                 if (functionDeclaration.body && functionDeclaration.body.kind === SyntaxKind.Block) {
                     // Proper function declarations can only have identifier names
-                    if (forEach((<Block>functionDeclaration.body).statements,
-                        s => s.kind === SyntaxKind.FunctionDeclaration && !isEmpty((<FunctionDeclaration>s).name.text))) {
-
+                    if (hasNamedFunctionDeclarations((<Block>functionDeclaration.body).statements)) {
                         return true;
                     }
 
-                    // Or if it is not parented by another function.  i.e all functions
-                    // at module scope are 'top level'.
+                    // Or if it is not parented by another function (except for parent functions that
+                    // are methods and constructors). I.e all functions at module scope are 'top level'.

```

Comment:
Functions in function are considered in the above if statement:

```ts
if (hasNamedFunctionDeclarations((<Block>functionDeclaration.body).statements)) {
    return true;
}
```
Just note that. The current implementation, `top-level` functions are just a classification of functions. They are not actual top-level functions. This follows the previous logic with nested functions. If a nested function has a function declarations inside, even though itself is nested. It is classified as `top-level`.

Though the comment here can probably be improved.